### PR TITLE
feat: update course statuses on publisher

### PR DESCRIPTION
**Pull request by Ali-D-Akbar**
[PROD-3218](https://2u-internal.atlassian.net/browse/PROD-3218)
Adds archive status to be used in Publisher home page.

Screenshot 2023-04-04 at 6 17 35 AM

### CODE REVIEW

File: course_discovery/apps/course_metadata/models.py [hunk 1]
```diff
@@ -1498,10 +1498,14 @@ def course_run_statuses(self):
 
         Note that it skips hidden and archived courses - this list is typically used for presentational purposes.
```
Comment (DawoudSheraz) [inline course_discovery/apps/course_metadata/models.py]: please update docstring.

File: course_discovery/apps/course_metadata/models.py [hunk 2]
```diff
@@ -1498,10 +1498,14 @@ def course_run_statuses(self):
...
         """
+        statuses = []
         now = datetime.datetime.now(pytz.UTC)
-        runs = self.course_runs.exclude(hidden=True).exclude(status=CourseRunStatus.Unpublished, end__lt=now)
-        statuses = runs.values_list('status', flat=True).distinct().order_by('status')
-        return list(statuses)
+        runs = self.course_runs.exclude(hidden=True)
+        if runs.filter(status=CourseRunStatus.Unpublished, end__lt=now).exists():
+            statuses = ['archived']
```
Comment (DawoudSheraz) [inline course_discovery/apps/course_metadata/models.py:1505]: do we need to update how this status is returned in API + ElasticSearch document? cause on course run level, this status will be unpublished and not archived. Just aligning expectations here before this is merged.

Comment (Ali-D-Akbar) [reply course_discovery/apps/course_metadata/models.py:1505]: Neither ES have a `course_run_statuses` field nor we're changing the `status` field in API. The data is already in sync for both API and ES.

Comment (Ali-D-Akbar) [reply course_discovery/apps/course_metadata/models.py:1505]: That reminded me of updating the course status filter too. Lemme do that quickly.

**Review by AfaqShuaib09 — APPROVED** [review]

**Review by DawoudSheraz — APPROVED** [review]

**Pull request merged**